Mage.Sets/src/mage/cards/r/ReturnThePast.java

```diff
@@ -0,0 +1,73 @@
+package mage.cards.r;
+
+import mage.abilities.Ability;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.effects.ContinuousEffectImpl;
+import mage.abilities.keyword.FlashbackAbility;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.players.Player;
+
+import java.util.UUID;
+
+/**
+ *
+ * @author notgreat
+ */
+public final class ReturnThePast extends CardImpl {
+
+    public ReturnThePast(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{4}{R}{R}");
+
+
+        // As long as it's your turn, each instant and sorcery card in your graveyard has flashback. Its flashback cost is equal to its mana cost.
+        this.addAbility(new SimpleStaticAbility(new ReturnThePastEffect()));
+    }
+
+    private ReturnThePast(final ReturnThePast card) {
+        super(card);
+    }
+
+    @Override
+    public ReturnThePast copy() {
+        return new ReturnThePast(this);
+    }
+}
+
+//Based on LierDiscipleOfTheDrownedFlashbackEffect
+class ReturnThePastEffect extends ContinuousEffectImpl {
```

Comment:
Not for current PR: there are another 3 cards with same abilities -- so it can use shared code (3 with same and "return to past" with my turn conditional).

![shot_240620_091611](https://github.com/magefree/mage/assets/8344157/b1b7217a-5301-4fe2-9768-4ab6ea7e6521)